### PR TITLE
Clean up utils tests.

### PR DESCRIPTION
**Pull request by tabergma**
**Proposed changes**:
- create a test file for each file in `rasa/utils`
- remove not used utils methods
- move all tests related to methods in `rasa/utils` to corresponding file in `tests/utils`
- no new functionality or tests were added

In a next PR we should move utils methods like `rasa.nlu.utils.list_files()` to the utils package.
A lot of methods in the utils package are not tested at all. We need to add some tests in a next PR.

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

File: rasa/utils/common.py [hunk 1]
```diff
@@ -53,41 +53,6 @@ def read_global_config() -> Dict[Text, Any]:
         return {}
 
 
-def write_global_config_value(name: Text, value: Any) -> None:
```
Comment (wochinge) [inline rasa/utils/common.py]: both functions are actually used by `rasa-x` and indirectly used e.g. in `check_license_and_metrics`

Comment (wochinge) [reply rasa/utils/common.py]: I think we should keep it in `rasa` btw. There might be some other information we want to store in their in the future (e.g. some default parameters)

File: rasa/utils/io.py [hunk 2]
```diff
@@ -234,18 +234,6 @@ def create_path(file_path: Text):
         os.makedirs(parent_dir)
 
 
-def zip_folder(folder: Text) -> Text:
```
Comment (wochinge) [inline rasa/utils/io.py]: that's also used by `rasa-x` but I think we can delete it nevertheless and refactor that part in rasa-x (would have to look in it a bit more)

File: tests/utils/test_io.py [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+    r = io_utils.read_yaml(config_with_env_var)
+    assert r["user"] == "user" and r["password"] == "pass"
+
+
+def test_read_yaml_string_with_multiple_env_vars_per_line():
+    config_with_env_var = """
+    user: ${USER_NAME} ${PASS}
+    password: ${PASS}
+    """
+    r = io_utils.read_yaml(config_with_env_var)
```
Comment (wochinge) [inline tests/utils/test_io.py]: @arthurTemporim had some comments about the naming of `r` . If you have time you could give that a proper name, e.g. `parsed` or `result`

File: rasa/utils/common.py (hunk 1 above)
Comment (tabergma) [reply rasa/utils/common.py]: good to know. will add them again.

File: rasa/utils/io.py (hunk 2 above)
Comment (tabergma) [reply rasa/utils/io.py]: ok. but maybe we keep it for now, because we first should refactor rasa-x then. will create an issue for that.

Comment (wochinge) [reply rasa/utils/io.py]: ok 👍

**Pull request merged**